daemon/api_systems.go

```diff
@@ -82,3 +89,44 @@ func getSystems(c *Command, r *http.Request, user *auth.UserState) Response {
 	}
 	return SyncResponse(&rsp, nil)
 }
+
+type systemActionRequest struct {
+	Action string `json:"action"`
+	client.SystemAction
+}
+
+func postSystemsAction(c *Command, r *http.Request, user *auth.UserState) Response {
+	var req systemActionRequest
+
+	systemLabel := muxVars(r)["label"]
+	if systemLabel == "" {
+		return BadRequest("system action requires the system label to be provided")
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&req); err != nil {
+		return BadRequest("cannot decode request body into system action: %v", err)
+	}
+	if decoder.More() {
+		return BadRequest("extra content found in request body")
+	}
+	if req.Action != "do" {
+		return BadRequest("unsupported action %q", req.Action)
+	}
+	if req.Mode == "" {
+		return BadRequest("system action requires the mode to be provided")
+	}
+
+	sa := devicestate.SystemAction{
+		Title: req.Title,
+		Mode:  req.Mode,
+	}
+	if err := c.d.overlord.DeviceManager().RequestSystemAction(systemLabel, sa); err != nil {
+		if os.IsNotExist(err) {
+			return NotFound("requested seed system %q does not exist", systemLabel)
+		}
+		// distinguish bad action?
```

Comment:
If you edit this anyway, maybe adding a "XXX" ?

Reply:
actually I tweaked this to return a known error